var/spack/repos/builtin/packages/krb5/package.py

```diff
@@ -80,9 +80,10 @@ def configure_args(self):
         return args
 
     def setup_build_environment(self, env):
-        env.prepend_path("LD_LIBRARY_PATH", self.spec["gettext"].prefix.lib)
+        if "intl" in self.spec["gettext"].libs.names:
+            env.prepend_path("LD_LIBRARY_PATH", "-lintl")
 
     def flag_handler(self, name, flags):
         if name == "ldlibs":
-            flags.append("-lintl")
+            flags.append(self.spec["gettext"].libs[4])
```

Comment:
Any way to avoid the hardwired number, here?

Reply:
Yes sorry, this is a vestige of my previous iteration of trying to solve this.  This should be the same test before adding -lintl as thet others.